Solutions/Marain.UserNotifications.Storage.AzureStorage/Marain/UserNotifications/Storage/AzureStorage/Template/AzureBlobTemplateStore.cs

```diff
@@ -70,32 +72,34 @@ public async Task<T> CreateOrUpdate<T>(string notificationType, CommunicationTyp
             this.logger.LogDebug("CreateOrUpdate: Storing template for notification type ", notificationType);
 
             // Gets the blob reference by the NotificationType
-            CloudBlockBlob blockBlob = this.blobContainer.GetBlockBlobReference(this.GetBlockBlobName(notificationType, communicationType));
+            BlobClient blockBlob = this.blobContainer.GetBlobClient(this.GetBlockBlobName(notificationType, communicationType));
 
             bool exists = await blockBlob.ExistsAsync().ConfigureAwait(false);
```

Comment:
This line should have been removed, because a) nothing looks at `exists` any more, and b) the test for existence is now taking place as part of the conditional update request.